File: src/cs5400-artificial-intelligence/puzzle-series/2018-sp-a-puzzle_1-isgx2/src/timer.py
#!/usr/local/bin/python3
#
# timer.py
# puzzle-assignment
#
# Created by Illya Starikov on 01/24/18.
# Copyright 2018. Illya Starikov. All rights reserved.
#

from datetime import datetime


class Timer():
    __start_time, __stop_time = None, None

    def start(self):
        """Starts the timer"""
        self.__start_time = datetime.now()

    def stop(self):
        """Stops the timer"""
        self.__stop_time = datetime.now()

    @property
    def difference(self):
        """Calculate the difference between the start time and the stop time.

        Returns:
            float: The amount of seconds that have pass between the timer starting
            and the timer stopping.
        """
        delta = self.__stop_time - self.__start_time
        return delta.total_seconds()

File: src/cs5400-artificial-intelligence/game-series/2018-sp-a-game-4-isgx2/Joueur.cpp/games/chess/chess-ai/src/timer.cpp
//
//  timer.cpp
//  Desktop
//
//  Created by Illya Starikov on 04/03/18.
//  Copyright 2018. Illya Starikov. All rights reserved.
//

#include "timer.h"

using namespace ChessEngine;

void Timer::start() {
    begin = std::chrono::steady_clock::now();
}

void Timer::stop() {
    end = std::chrono::steady_clock::now();
}

double Timer::elapsed() {
    static const double MILLION = 1000000.0;
    return std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::steady_clock::now() - begin).count() / MILLION;
}

